frontend/app/utils/date-utils.ts

```diff
@@ -51,3 +51,76 @@ export function getLocalizedMonths(
     text: formatter.format(Date.UTC(0, i, 1)), // formatted month name
   }));
 }
+
+/**
+ * Validates whether a given string is in the format YYYY-MM-DD.
+ *
+ * @param date - The date string to be validated.
+ * @returns Returns `true` if the string is a valid date in the format YYYY-MM-DD, otherwise `false`.
+ *
+ * @example
+ * ```javascript
+ * const validDate = "2023-05-20";
+ * const invalidDate = "20-05-2023";
+ *
+ * console.log(isValidDateString(validDate)); // true
+ * console.log(isValidDateString(invalidDate)); // false
+ * ```
+ */
+export const isValidDateString = (dateString: string) => {
+  const dateRegex = /^\d{4}-\d{2}-\d{2}$/;
+  if (!dateRegex.test(dateString)) {
+    return false;
+  }
+  const date = new Date(dateString);
+  return !isNaN(date.getTime());
+};
+
+/**
+ * Validates whether a given string is in past or today.
+ *
+ * @param date - The date string to be validated.
+ * @returns Returns `true` if the string is a valid date in the format YYYY-MM-DD and the date is in past or today's date, otherwise `false`.
+ *
+ * This function first validates the input date string using the `isValidDateString` function.
+ * If the date string is invalid, an error is thrown. Otherwise, it compare date
+ * object with the date string set to the start of the day in UTC (00:00:00.000Z).
+ *
+ * @example
+ * ```javascript
+ * const validDateInPaste = "2023-05-20";
+ * const invalidDateInFuture = "6078-05-20";
+ *
+ * console.log(isDateInPastOrToday(validDateInPaste)); // true
+ * console.log(isDateInPastOrToday(invalidDateInFuture)); // false
+ * ```
+ */
+export function isDateInPastOrToday(dateString: string): boolean {
+  if (!isValidDateString(dateString)) {
+    throw new Error('Invalid date format. Expected YYYY-MM-DD.');
+  }
+  const inputDate = new Date(dateString);
+
+  const dateFormatEST = new Intl.DateTimeFormat('en-CA', {
+    timeZone: 'America/Toronto',
+    year: 'numeric',
+    month: '2-digit',
+    day: '2-digit',
+  });
+
+  const maxAllowedDate = new Date(`${dateFormatEST.format(new Date())}T00:00:00`);
+
+  if (inputDate <= maxAllowedDate) return true;
+  return false;
```

Comment:
Use [isPast](https://date-fns.org/v4.1.0/docs/isPast) and [isToday](https://date-fns.org/v4.1.0/docs/isToday) from `date-fns`.

```suggestion
  const inputDate = parseISO(dateString);
  return isPast(inputDate) || isToday(inputDate);
```

Reply:
We can't use `isPast()` because we have to normalize for a specific timezone. Instead, you have to do something like this:

``` ts
isBefore(inputDate, getStartOfDayInTimezone('Canada/Eastern'))
```